openedx/features/philu_courseware/tests/test_competency_assessment_scores_api.py

```diff
@@ -0,0 +1,150 @@
+
+    def test_un_enrolled_user_request(self):
+        """Test unenrolled user request"""
+        response = self.get_competency_assessment_api_response(self.create_chapter().location.block_id)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_unauthorized_request(self):
+        """Test unauthorized request"""
+        self.client.force_authenticate(user=None)
+        response = self.get_competency_assessment_api_response(self.create_chapter().location.block_id)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class PrePostAssessmentScoreForEnrolledUserTest(PrePostAssessmentMixin, APITestCase):
+    """Tests for enrolled users"""
+
+    def setUp(self):
+        super(PrePostAssessmentScoreForEnrolledUserTest, self).setUp()
+        self.create_course_and_user()
+        self.client.force_authenticate(self.user)
+        CourseEnrollmentFactory.create(user=self.user, course_id=self.course.id)
+
+    def test_invalid_chapter_id(self):
+        """Test api for invalid chapter id"""
+        response = self.get_competency_assessment_api_response('invalid_chapter_id')
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_no_competency_assessment_in_chapter(self):
+        """Test if module/chapter has no pre/post assessment"""
+        chapter = self.create_chapter()
+        ItemFactory.create(
+            parent_location=chapter.location,
+            display_name='Welcome',
+            category='sequential'
+        )
+        response = self.get_competency_assessment_api_response(chapter.location.block_id)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_only_pre_assessment_chapter(self):
+        """Test module/chapter has only pre assessment"""
+        chapter = self.create_chapter()
+        self.create_assessment_subsection(chapter)
+        response = self.get_competency_assessment_api_response(chapter.location.block_id)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_only_post_assessment_chapter(self):
+        """Test module/chapter has only post assessment"""
+        chapter = self.create_chapter()
+        self.create_assessment_subsection(chapter, constants.POST_ASSESSMENT_FORMAT)
+        response = self.get_competency_assessment_api_response(chapter.location.block_id)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_competency_assessment_chapter(self):
+        """Test module/chapter has both pre and post assessments"""
+        chapter = self.create_chapter()
+        self.create_assessment_subsection(chapter)
+        self.create_assessment_subsection(chapter, constants.POST_ASSESSMENT_FORMAT)
+        response = self.get_competency_assessment_api_response(chapter.location.block_id)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```

Comment:
please check , is there is any way to reduce these assert checks ?

Reply:
I've checked that there are cases in our unit tests where we use multiple asserts in a single unit test